src/containers/flows/TransactionWizards/TransactionWizard.tsx

```diff
@@ -0,0 +1,361 @@
+	const { getBidsForDomain } = useBidProvider();
+
+	const { wildPriceUsd } = useCurrency();
+
+	const [currentStep, setCurrentStep] = useState(0);
+	const [domainMetadata, setDomainMetadata] = useState<Metadata | undefined>();
+
+	const [isTransactionFailed, setTransactionFailed] = useState(false);
+
+	const [isRejectModalOpenned, setRejectModalOpenned] = useState(false);
+
+	const [currentHighestBid, setCurrentHighestBid] = useState<Bid | undefined>();
+
+	const [isWaletLoading, setWalletLoading] = useState(false);
+	const [isTransactionProccessing, setTransactionProccessing] = useState(false);
+
+	//- Web3 Wallet Data
+	const walletContext = useWeb3React<Web3Provider>();
+	const { account, library } = walletContext;
+
+	const [zauctionApproved, isApprovalLoading] = useZauctionApproval({
+		library,
+		domainId: domain.id,
+		account,
+	});
+
+	const [currentBuyPrice] = useBuyNowPrice({
+		library,
+		account,
+		domainId: domain.id,
+	});
+
+	//////////////
+	// Handlers //
+	//////////////
+
+	const handleCancel = () => {
+		steps[currentStep].actions
+			.cancel()
+			.then(() => {
+				setRejectModalOpenned(true);
+			})
+			.catch(() => {
+				setTransactionFailed(true);
+			});
+	};
+
+	const handleNextStep = async () => {
+		setWalletLoading(true);
+
+		steps[currentStep].actions
+			.next()
+			.then(async (transaction) => {
+				if (typeof transaction !== 'undefined') {
+					setWalletLoading(false);
+					setTransactionProccessing(true);
+					await transaction
+						.wait()
+						.then(() => {
+							setCurrentStep(currentStep + 1);
+						})
+						.catch((error: any) => {
+							console.error('network error, unable to set buy price');
+							console.error(error);
+						})
+						.finally(() => {
+							setTransactionProccessing(false);
+						});
+				} else {
+					setCurrentStep(currentStep + 1);
+					setWalletLoading(false);
+				}
+			})
+			.catch((error) => {
+				console.warn(error);
+				setTransactionFailed(true);
+				setWalletLoading(false);
+			});
+	};
+
+	const handleTransactionSuccess = () => {
+		successHandler();
+	};
+
+	const handleTransactionCancel = () => {
+		cancelHandler();
+	};
+
+	/////////////
+	// Effects //
+	/////////////
+
+	useEffect(() => {
+		console.log(zauctionApproved);
+		if (zauctionApproved && !isApprovalLoading) {
+			setCurrentStep((prevStep) => prevStep + 1);
+		}
+	}, [zauctionApproved, isApprovalLoading]);
+
+	useEffect(() => {
+		let isSubscribed = true;
+
+		const getCurrentHighestBid = async () => {
+			// Get highest bid
+			const allBids = await getBidsForDomain(domain);
+
+			if (!allBids || allBids.length === 0) {
+				return;
+			}
+			const highestBid = allBids.reduce(function (prev, current) {
+				return prev.amount > current.amount ? prev : current;
+			});
+
+			if (isSubscribed) {
+				setCurrentHighestBid(highestBid);
+			}
+		};
+
+		getCurrentHighestBid();
+
+		return () => {
+			isSubscribed = false;
+		};
+	}, [domain, getBidsForDomain]);
+
+	useEffect(() => {
```

Comment:
Idea to implement Transaction Wizard is good but I think we shouldn't write these effects related to different scenarios in one place to avoid confusion and tightly coupling.